tests/test_slots.py

```diff
@@ -0,0 +1,25 @@
+from falcon import Request, Response
+import falcon.testing as testing
+
+
+class TestSlots(testing.TestBase):
+
+    def test_slots_request(self):
+        env = testing.create_environ()
+        req = Request(env)
+        raised = False
+        try:
+            req.doesnt = 'exist'
+        except AttributeError:
+            raised = True
+        self.assertFalse(raised, 'Unable to add to __slots__ dynamically')
+
+    def test_slots_response(self):
+        resp = Response()
+        resp.doesnt = 'exist'
+        raised = False
+        try:
+            resp.doesnt = 'exist'
+        except AttributeError:
+            raised = True
```

Comment:
What do you think about just calling `self.fail()` here instead of using the `raised` variable?